js/therock.js

```diff
@@ -1251,6 +1253,92 @@ module.exports = class therock extends Exchange {
         return this.parseOrder (response);
     }
 
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name exmo#fetchOHLCV
+         * @description fetches historical candlestick data containing the open, high, low, and close price, and the volume of a market
+         * @param {str} symbol unified symbol of the market to fetch OHLCV data for
+         * @param {str} timeframe the length of time each candle represents in minutes
+         * @param {int|undefined} since timestamp in ms of the earliest candle to fetch
+         * @param {int|undefined} limit the maximum amount of candles to fetch
+         * @param {dict} params extra parameters specific to the exmo api endpoint
+         * @returns {[[int]]} A list of candles ordered as timestamp, open, high, low, close, volume
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const period = this.convertTimeframeToMinutes (timeframe);
+        const request = {
+            'id': market['id'],
+            'period': period,
+        };
+        if (since === undefined) {
+            request['after'] = this.iso8601 (since);
+        }
+        const response = await this.publicGetFundsIdOhlcStatistics (this.extend (request, params));
+        //
+        //     [
+        //         {
+        //             "fund_id": "BTCUSDT",
+        //             "open": 31500.0,
+        //             "high": 31500.0,
+        //             "low": 31500.0,
+        //             "close": 31500.0,
+        //             "average": 31500.0,
+        //             "weighted_average": 31500.0,
+        //             "base_volume": 0.0,
+        //             "traded_volume": 0.0,
+        //             "interval_starts_at": "2022-06-06T16:40:00.000Z",
+        //             "interval_ends_at": "2022-06-06T16:50:00.000Z"
+        //         }
+        //         ...
+        //     ]
+        //
+        return this.parseOHLCVs (response, market, timeframe, since, limit);
+    }
+
+    convertTimeframeToMinutes (timeframe) {
```

Comment:
We have a `parseTimeframe` base method for this purpose that returns the number of seconds corresponding to the specified timeframe. I suggest that we use `parseTimeframe` instead of `convertTimeframeToMinutes`.

Reply:
Also, we will still have to add the `timeframes` property to the `describe()` function (like we do in all other exchanges, otherwise the users don't know which timeframes are available). We can take their most used timeframes from their website GUI, that would make a list of "standard" timeframes for this exchange that is more or less aligned with the other exchanges. Ideally we should always use a set of predefined strings from `this.timeframes` dictionary, and also it would be great if we could support recalculating the non-standard timeframes with `this.parseTimeframe` if the user requests some non-standard timeframes like `7m` or `13m`.